crates/bevy_ecs/src/query/state.rs

```diff
@@ -83,10 +86,47 @@ pub struct QueryState<D: QueryData, F: QueryFilter = ()> {
     pub(super) is_dense: bool,
     pub(crate) fetch_state: D::State,
     pub(crate) filter_state: F::State,
+    sync_by_observer: bool,
     #[cfg(feature = "trace")]
     par_iter_span: Span,
 }
 
+impl<D: QueryData + 'static, F: QueryFilter + 'static> Component for QueryState<D, F> {
```

Comment:
Why do a manual implementation of `Component`?  You can specify hooks using the derive macro [with a `#[component(on_add = on_add_function)]` attribute](https://docs.rs/bevy/latest/bevy/ecs/component/trait.Component.html#adding-components-hooks).